### PR TITLE
add hqmusic a Vietnamese private site. resolves #13896

### LINKED ISSUE #13896: [req]: HQMusic
### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Is there already a request for your tracker?

- [X] I have checked older issues, open and closed

### Type

Private

### Tracker details

```markdown
- Name: HQMusic
- URL: https://tracker.hqmusic.info/
- Description: HQMusic is a VIETNAMESE Private Torrent Tracker for MUSIC
- Minimum Seeding Time and Minimum Ratio (if Private): MR: 0.4 MST: 2.5d
- Additional Comments: 
Engine: NexusPHP Standard v1.5 Beta 4
Registration is currently invite only
Account Inactivity:
Inactive accounts will be deleted according to the following rules:
  1. Power Users and above will never have their accounts deleted.
  2. Users and above will never have their accounts deleted if they park their accounts (See [User CP]).
  3. Parked accounts will be deleted if they do not log in within 365 days.
  4. Unparked accounts will be deleted if they do not log in within 240 days.
  5. Inactive accounts downloading or uploading on the tracker will be deleted after 240 days of not logging in or after 120 days from the time of registration.
```

### PR DESCRIPTION
**Pull request by ilike2burnthing**
resolves #13896

Used jpopsuki as a base, not sure if anything here can be used for it.

Annoyingly, what is shown on the site and what is returned to Jackett are different, so we're missing actual download links and tags:

<details>
<summary>Site:</summary>

```html
<tr class="torrent">
		<td class="center cats_col">
			<div title="Pop" class="tooltip cats_music tags_pop"></div>
		</td>
		<td class="big_info">
			<div class="group_image float_left clear">
						<img src="http://i.imgur.com/kovfq8Em.jpg" width="90" height="90" alt="Cover" onclick="lightbox.init('http://i.imgur.com/kovfq8E.jpg', 90)">
			</div>
			<div class="group_info clear">
				<span>
					[ <a href="torrents.php?action=download&amp;id=265&amp;authkey=REMOVED&amp;torrent_pass=REMOVED" class="tooltip" title="Download">DL</a>
					| <a href="reportsv2.php?action=report&amp;id=265" class="tooltip" title="Report">RP</a> ]
				</span>
				<a href="artist.php?id=154" dir="ltr">Tong Li</a> - <a href="torrents.php?id=273&amp;torrentid=265#torrent265" class="tooltip" title="View torrent" dir="ltr">600 Sec</a> [2007] [Album]				<div class="torrent_info">FLAC / Lossless / Log (100%) / Cue / CD / <strong class="torrent_label tooltip tl_free tl_neutral" title="Neutral Leech!" style="white-space: nowrap;">Neutral Leech!</strong></div>
				<div class="tags"><a href="torrents.php?action=basic&amp;taglist=pop">pop</a>, <a href="torrents.php?action=basic&amp;taglist=folk">folk</a>, <a href="torrents.php?action=basic&amp;taglist=chinese">chinese</a>, <a href="torrents.php?action=basic&amp;taglist=classic">classic</a>, <a href="torrents.php?action=basic&amp;taglist=freeleech">freeleech</a></div>
			</div>
		</td>
		<td>35</td>
		<td class="nobr"><span class="time tooltip" title="Oct 17 2016, 12:45">6 years ago</span></td>
		<td class="number_column nobr">464.63 MB</td>
		<td class="number_column">38</td>
		<td class="number_column">2</td>
		<td class="number_column">0</td>
	</tr>
```
</details>

<details>
<summary>Jackett:</summary>

```html
<tr class="torrent">
	<td class="center cats_col">
		<div title="Pop" class="tooltip cats_music tags_pop"></div>
	</td>
	<td class="big_info">
		<div class="group_image float_left clear">
			<img src="http://i.imgur.com/kovfq8Em.jpg" width="90" height="90" alt="Cover" onclick="lightbox.init('http://i.imgur.com/kovfq8E.jpg', 90)">
		</div>
		<div class="group_info clear">
			
			<a href="artist.php?id=154" dir="ltr">Tong Li</a>
			-
			<a href="torrents.php?id=273&amp;torrentid=265#torrent265" class="tooltip" title="View torrent" dir="ltr">600 Sec</a>
			[2007] [Album]
			<div class="torrent_info">FLAC / Lossless / Log (100%) / Cue / CD /
				<strong class="torrent_label tooltip tl_free tl_neutral" title="Neutral Leech!" style="white-space: nowrap;">Neutral Leech!</strong>
			</div>
			
		</div>
	</td>
	<td>35</td>
	<td class="nobr">
		<span class="time tooltip" title="Oct 17 2016, 12:45">6 years ago</span>
	</td>
	<td class="number_column nobr">464.63 MB</td>
	<td class="number_column">38</td>
	<td class="number_column">2</td>
	<td class="number_column">0</td>
</tr>
```
</details>

I had a play about with some account settings, but no luck. May need some header passed, not sure. @garfield69 anything obvious jump out to you? It's not that important, but I spent too long trying to work it out to not ask.

### CODE REVIEW

**Comment by ilike2burnthing** [comment]
> PS: you left the authkey/torrent_pass in the examples.

Removed, ta.

**Comment by garfield69** [comment]
I'll take a look later, but for now, have you looked at indietorrents, its a project gazelle yaml indexer too.

**Comment by ilike2burnthing** [comment]
> Check the cookies in browser and try testing it with a normal user-agent.

Tried switching over to cookie and UA, even threw in all the other request headers as well, no change.

> Btw their API seems to be working at /ajax.php?action=browse&order_by=time&order_way=desc&group_results=1
EDIT: now I've seen your comment in the yml file, check it with group_results=1

Ah balls, it works... where's the whisky!?

**Comment by garfield69** [comment]
htmldecode filter 578de3ca42ad1b25814bb2b12e2f88fa699154a2 https://github.com/Jackett/Jackett/wiki/Definition-format#htmldecode

File: src/Jackett.Common/Definitions/hqmusic.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        - name: re_replace
+          args: ["^0$", ""]
+    _hasCue:
+      selector: hasCue
+    _media:
+      selector: media
+    title:
+      # John Mayer - Born And Raised [2012] [Album] FLAC / Lossless / Log (100%) / Cue / CD
+      # Duy - I Go [2022] [Single] Other / Lossless / WEB
+      text: "{{ .Result.artist }} - {{ .Result._groupName }} {{ if .Result.year }}[{{ .Result.year }}] {{ else }}{{ end }}[{{ .Result._releaseType }}] {{ .Result._format }} / {{ .Result._encoding }} {{ if eq .Result._hasLog \"True\" }}/ Log {{ else }}{{ end }}{{ if .Result._logScore }}({{ .Result._logScore }}%){{ else }}{{ end }}{{ if eq .Result._hasCue \"True\" }} / Cue{{ else }}{{ end }} / {{ .Result._media }}"
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hqmusic.yml:134]: Time well spent... 🙄

Comment (ilike2burnthing) [reply src/Jackett.Common/Definitions/hqmusic.yml:134]: ...brb

Comment (ilike2burnthing) [reply src/Jackett.Common/Definitions/hqmusic.yml]: Done 😉

**Comment by ilike2burnthing** [comment]
The comment from @mynameisbogdan disappeared, so I'll just answer here.

No, I purposefully didn't change `year` as we don't want `0` to be returned as a valid result that could be used outside of the title:

https://github.com/Jackett/Jackett/blob/200d9dd37668c3e144b808cc870404729472812a/src/Jackett.Common/Models/ResultPage.cs#L106

The rest can't be used elsewhere, so they're fine.

@garfield69 this is ready for review on my end. Feel free to make any changes.

File: src/Jackett.Common/Definitions/hqmusic.yml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  inputs:
+    username: "{{ .Config.username }}"
+    password: "{{ .Config.password }}"
+    keeplogged: 1
+    login: "Log In"
+  error:
+    - selector: form#loginform > span.warning
+  test:
+    path: torrents.php
+
```
[the remark this answers is not in the document]
Comment (ilike2burnthing) [reply src/Jackett.Common/Definitions/hqmusic.yml:58]: No issue with this, but is there any benefit/difference? The same torrent file is downloaded either way.

**Review by garfield69 — APPROVED** [review]
Awesome work.

**Pull request merged**